core/src/main/java/com/linecorp/armeria/server/metric/PrometheusExpositionService.java

```diff
@@ -85,11 +90,30 @@ public PrometheusExpositionService(CollectorRegistry collectorRegistry) {
 
     @Override
     protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
-        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
-        try (OutputStreamWriter writer = new OutputStreamWriter(stream)) {
-            TextFormat.write004(writer, collectorRegistry.metricFamilySamples());
-        }
-        return HttpResponse.of(HttpStatus.OK, CONTENT_TYPE_004, stream.toByteArray());
+        final HttpResponseWriter responseWriter = HttpResponse.streaming();
+        responseWriter.write(ResponseHeaders.builder(HttpStatus.OK).contentType(CONTENT_TYPE_004).build());
+        ctx.blockingTaskExecutor().execute(() -> {
+            try (BufferedWriter writer = new BufferedWriter(new Writer() {
+                @Override
+                public void write(char[] cbuf, int off, int len) {
+                    responseWriter.write(HttpData.ofUtf8(new String(cbuf, off, len)));
```

Comment:
Q) If a `ClosedSessionException` is thrown, what is the behavior?

My guess is that `BufferedWriter` will be closed, consequently calling `responseWriter.close();` (so the exception isn't propagated to `responseWriter`)
Let me know if I'm misunderstanding 😅

Reply:
Maybe you meant `ClosedStreamException`?

Reply:
Ah yes 😅  My mistake

Reply:
Since `responseWriter` was closed already, we can propagate the exception to `responseWriter` anymore. The only thing we can do is abort the writing and silently ignore the late exception?

Reply:
Ah, right.  It seems like `ClosedStreamException` is thrown when the `responseWriter` is already closed. (in which case, further calls to `close()` or `close(e)` would be ignored anyways.

I think our default `blockingTaskExecutor` silently ignores exceptions thrown, so I think you can ignore this comment 🙇